build-system/tasks/serve.js

```diff
@@ -81,15 +113,26 @@ function getMiddleware() {
 /**
  * Launches a server and waits for it to fully start up
  *
- * @param {?Object} connectOptions
- * @param {?Object} serverOptions
- * @param {?Object} modeOptions
+ * @param {?ConnectOptionsDef} connectOptions
+ * @param {?ServerOptionsDef} serverOptions
+ * @param {?ModeOptionsDef} modeOptions
  */
 async function startServer(
   connectOptions = {},
   serverOptions = {},
   modeOptions = {}
 ) {
+  if (
+    'new_server' in serverOptions ? serverOptions.new_server : argv.new_server
+  ) {
+    try {
+      buildNewServer();
```

Comment:
I see that in #30681, you added a call to `buildNewServer()` to `integration.js`. I think that change should be reverted because the server startup logic should entirely live in `server.js`. Other tasks should trust the server to start itself correctly. Can we clean that up too?

Reply:
agree with the statement, but `buildNewServer` isnt really start up logic and more compile logic. the code essentially isnt there (since its in typescript) before we make this call. `buildNewServer` i believe should be done before the task, i just placed it there since its the latest we can call it to make "lazy"

Reply:
do you think it would be imperative i make this change here, or can i follow up on a separate PR? I am working on it already but i wasn't sure if i could clean separate the issue out in another PR

Reply:
#32891 makes this change and cleans up `integration.js`. Once reviewed and merged, we shouldn't need any of these changes.